pkg/services/featuremgmt/openfeature.go

```diff
@@ -40,13 +51,16 @@ func newOpenFeatureService(pType string, u *url.URL, staticFlags map[string]bool
 
 	client := openfeature.NewClient("grafana-openfeature-client")
 	return &OpenFeatureService{
-		log:      log.New("openfeatureservice"),
-		provider: p,
-		Client:   client,
+		log:                      log.New("openfeatureservice"),
+		provider:                 p,
+		stackID:                  stackID,
+		Client:                   client,
+		httpClientProvider:       httpClientProvider,
+		signerMiddlewareProvider: signerMiddlewareProvider,
 	}, nil
 }
 
-func createProvider(providerType string, u *url.URL, staticFlags map[string]bool) (openfeature.FeatureProvider, error) {
+func createProvider(stackID string, providerType string, u *url.URL, staticFlags map[string]bool, httpClientProvider *sdkhttpclient.Provider, signerMiddlewareProvider *middleware.CloudAccessPolicyTokenSignerMiddlewareProvider) (openfeature.FeatureProvider, error) {
```

Comment:
The `stackID` parameter implies that a new provider is created per stack. That won't work. We need a single HTTP client that can handle many stacks/tenants - that information needs to come at runtime from the context...

Reply:
Addressed.